src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java

```diff
@@ -2122,6 +2163,25 @@ else if (null != sourceCryptoMeta) {
 
                 case microsoft.sql.Types.MONEY:
                 case microsoft.sql.Types.SMALLMONEY:
+                    if (null == colValue) {
+                        writeNullToTdsWriter(tdsWriter, bulkJdbcType, isStreaming);
+                    } else {
+                        /*
+                         * if the precision that user provides is smaller than the precision of the actual value, the
+                         * driver assumes the precision that user provides is the correct precision, and throws
+                         * exception
+                         */
+                        if (bulkPrecision < Util.getValueLengthBaseOnJavaType(colValue, JavaType.of(colValue), null,
+                                null, JDBCType.of(bulkJdbcType))) {
+                            MessageFormat form = new MessageFormat(
+                                    SQLServerException.getErrString("R_valueOutOfRange"));
+                            Object[] msgArgs = {SSType.DECIMAL};
```

Comment:
Shouldn't this be `destSSType` instead?

Reply:
Sorry, which part of this code should be destSSTyle?

Reply:
Actually, I understood. Yes, this should be destSSType instead. I've made similar changes to the decimal/numeric case as well because I think it applies there as well, too.